### PR TITLE
metrics: support strings

### PR DESCRIPTION
**Pull request by paulourbano**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/utils/diff.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def _diff_vals(old, new, with_unchanged):
     res = {"old": old, "new": new}
     if isinstance(new, (int, float)) and isinstance(old, (int, float)):
         res["diff"] = new - old
+    elif isinstance(new, str) and isinstance(old, str):
+        res["diff"] = ""
```
Comment (daavoo) [inline dvc/utils/diff.py:30]: I don't think this is really needed. The default behavior when `diff` entry is missing should be just fine, as a placeholder `-` gets rendered in the UI.

**Review by daavoo — CHANGES_REQUESTED** [review]
Thanks for the contribution @paulourbano 

Could you drop the `diff` change and update the tests accordingly?

**Comment by paulourbano** [comment]
> Thanks for the contribution @paulourbano !
> 
> Could you drop the `diff` change and update the tests accordingly?

Sure, I will do the changes.
Thanks!

**Comment by paulourbano** [comment]
Test failures (`tests/func/test_lockfile.py::test_cmd_changes_other_orders_are_preserved`, among others) are also shwoing in `main` with latest commit 1f03c40dbe47af8a7f64f6e1200aecd4b07273b0. Root cause seems not to be related to this PR.

**Review by daavoo — APPROVED** [review]

**Pull request merged**